framework/widgets/Pjax.php

```diff
@@ -133,6 +137,23 @@ public function init()
     }
 
     /**
+     * Custom ID generation for Pjax widgets
+     * This override is to attain a more reliable match between ID's of local and remote Pjax widgets (#12969)
+     * The implementation follows the solution proposed in the following comment:
+     * @link https://github.com/yiisoft/yii2/pull/12977#issuecomment-260123078
+     * @since 2.0.11
+     */
+    private static $pjaxCounter = 0;
+    private $_pjaxId;
+    public function getId($autoGenerate = true)
```

Comment:
There's no need to override this method. You just have to declare the following in Pjax class:

```php
 public static $counter = 0;
```